{compiler}[GCCcore/9.3.0] Add NVHPC v20.9, v20.11 and default to accept EULA

AndiH opened the pull request:
This pull request adds NVHPC 20.11, starting off of #11694.

Additionally, I got approval from @samcmill to default-accept the EULA and have, hence, uncommented the `source_urls` definition. I added a comment relating to the EULA, though.

lexming reviewed (commented):
@AndiH could you also add the `source_urls` and comment about the EULA in v20.11 to the easyconfig of v20.9?
I also have a minor suggestion to reduce duplicate code in the easyconfig and make it more readable. Please see below

boegel commented:
@AndiH W.r.t. the EULA, we could enhance the NVHPC easyblock to use the `EasyBlock.check_accepted_eula` method, see https://github.com/easybuilders/easybuild-framework/pull/3535?

AndiH commented:
@boegel We could! But since we got the general ok, I'd suggest to rather always accept it!

verdurin commented:
@AndiH any update on the suggestions from @lexming ?

AndiH commented:
@verdurin @lexming I think it makes it less readable (because more indirection), but easier to update in the future. I'll gladly implement. Give me a few days, though; I'm currently officially on parental leave :) .

AndiH commented:
I've added the following changes

* implemented @lexming's suggestions regarding restructuring of source filenames using `local_tarball_tmpl`
* backported these changes plus the EULA default-accept to NVHPC 20.7

lexming commented on easybuild/easyconfigs/n/NVHPC/NVHPC-20.11.eb:
```diff
@@ -0,0 +1,68 @@
+
+toolchain = SYSTEM
+
+# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
```
```suggestion
# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
accept_eula = True
```

lexming commented on easybuild/easyconfigs/n/NVHPC/NVHPC-20.9.eb:
```diff
@@ -0,0 +1,68 @@
+
+toolchain = SYSTEM
+
+# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
```
```suggestion
# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
accept_eula = True
```

lexming reviewed (changes requested):
@AndiH thanks for updating the PR. I made the changes requested by @boegel to the easyblock of NVHPC (see https://github.com/easybuilders/easybuild-easyblocks/pull/2311) to better handle the acceptance of the EULA.

Since you already checked that the EULA of NVHPC can be automatically accepted by just downloading the sources, we can add `accept_eula = True` to these easyconfigs to automatize the process. See below

AndiH commented:
Oh, sorry, I didn't recognize there was additional work for me to be done :). It's done now!

verdurin commented:
Looks fine.

verdurin commented:
@boegelbot please test @ generoso

lexming commented:
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node106.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/f8adb4fa839043d16d7df4def48a4fb0 for a full test report.

lexming reviewed (approved):
@AndiH thanks for updating the PR!
@verdurin your test failed because `NVHPC` requires some `--cuda-compute-capabilities` to be set
LGTM

lexming commented:
Going in, thanks @AndiH !
